src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -1038,71 +1105,87 @@ class SceneryWindow final : public Window
                 auto* tabInfo = GetSceneryTabInfoForGroup(sceneryGroupIndex);
                 if (tabInfo != nullptr)
                 {
-                    tabInfo->AddEntry(selection);
+                    tabInfo->AddEntry(selection, true);
                     return;
                 }
             }
 
             // If scenery is no tab, add it to misc
             if (!tabIndex.has_value())
             {
-                auto* tabInfo = GetSceneryTabInfoForGroup(OBJECT_ENTRY_INDEX_NULL);
+                auto* tabInfo = GetSceneryTabInfoForMisc();
                 if (tabInfo != nullptr)
                 {
-                    tabInfo->AddEntry(selection);
+                    tabInfo->AddEntry(selection, true);
                 }
             }
+
+            // Add all scenery to search tab
+            auto tabInfo = GetSceneryTabInfoForSearch();
+            if (tabInfo != nullptr)
+            {
+                tabInfo->AddEntry(selection, true);
+            }
+        }
+    }
+
+    u8string& GetFilterString(const size_t tabIndex)
+    {
+        if (tabIndex >= _filters.size())
+        {
+            _filters.resize(tabIndex + 1);
```

Comment:
I would use something else than a deque here since you are resizing and you can have unused locations in the vector. I would use an unordered_map so you only use memory you need.